crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -132,6 +160,18 @@ impl ProtocolFees {
             .collect_vec();
         boundary::order::to_domain(order, protocol_fees)
     }
+
+    /// Internal function in `ProtocolFees` in order to determine whether the
+    /// driver has to apple the protocol fee to an order/solution
+    fn driver_must_apply_protocol_fee(&self, address: H160) -> bool {
+        let is_address_whitelisted_for_full_discount = self
+            .protocol_fee_discounted_addresses
+            .get(&address)
+            .map(|kind| *kind == ProtocolFeeDiscountKind::Full)
+            .unwrap_or_default();
+
+        !is_address_whitelisted_for_full_discount
```

Comment:
nit: Seems easier to read than having to worry about negations and default values.
```suggestion
        let discount = self
            .protocol_fee_discounted_addresses
            .get(&address);

        discount != Some(&ProtocolFeeDiscountKind::Full)
```